pkg/deploy/controller/deploymentconfig/controller_test.go

```diff
@@ -1,835 +1,418 @@
-			expectedLatest:  true,
-			expectedErr:     false,
+			name:             "scale a completed deployment, previously synced",
+			replicas:         3,
+			newVersion:       1,
+			expectedReplicas: 3,
+			before: []deployment{
+				{version: 1, replicas: 1, replicasA: newint(1), status: deployapi.DeploymentStatusComplete, cancelled: false},
+			},
+			after: []deployment{
+				{version: 1, replicas: 3, replicasA: newint(3), status: deployapi.DeploymentStatusComplete, cancelled: false},
+			},
+			errExpected: false,
 		},
+		// TODO: fails
 		{
-			name:              "multiple errors",
-			controller:        controller,
-			version:           0,
-			hasMultipleErrors: true,
-			status:            deployapi.DeploymentStatusFailed,
-			reactions: []reaction{
-				{
-					verb:     "get",
-					resource: "imagestreamtags",
-					fn: func(action ktestclient.Action) (handled bool, ret runtime.Object, err error) {
-						return true, nil, kerrors.NewNotFound("imagestreamtag", "test-image-stream:latest")
-					},
-				},
-				{
-					verb:     "get",
-					resource: "imagestreamtags",
-					fn: func(action ktestclient.Action) (handled bool, ret runtime.Object, err error) {
-						return true, nil, kerrors.NewNotFound("imagestreamtag", "second-is:latest")
-					},
-				},
-				{
-					verb:     "list",
-					resource: "buildconfigs",
-					fn: func(action ktestclient.Action) (handled bool, ret runtime.Object, err error) {
-						return true, &buildapi.BuildConfigList{}, nil
-					},
-				},
-				{
-					verb:     "get",
-					resource: "imagestreams",
-					fn: func(action ktestclient.Action) (handled bool, ret runtime.Object, err error) {
-						return true, nil, nil
-					},
-				},
-			},
-			expectedDetails: "Deployment config \"config\" blocked by multiple errors:\n\n\t* The image trigger for image stream tag \"test-image-stream:latest\" will have no effect because image stream tag \"test-image-stream:latest\" does not exist.\n\t* The image trigger for image stream tag \"second-is:latest\" will have no effect because image stream tag \"second-is:latest\" does not exist.",
-			expectedErr:     false,
+			name:             "scale a completed deployment, multiple previously deployed and synced",
+			replicas:         2,
+			newVersion:       2,
+			expectedReplicas: 1,
+			before: []deployment{
+				{version: 1, replicas: 0, replicasA: newint(0), status: deployapi.DeploymentStatusComplete, cancelled: false},
+				{version: 2, replicas: 2, replicasA: newint(2), status: deployapi.DeploymentStatusComplete, cancelled: false},
+			},
+			after: []deployment{
+				{version: 1, replicas: 0, replicasA: newint(0), status: deployapi.DeploymentStatusComplete, cancelled: false},
+				{version: 2, replicas: 1, replicasA: newint(1), status: deployapi.DeploymentStatusComplete, cancelled: false},
+			},
+			errExpected: false,
```

Comment:
Oops, I think I misrepresented the scenario. I don't have time to redo it before Monday, but here's the manual test I was porting to the unit scenarios:

1. Create deploymentConfig(replicas=1,triggers=[])
2. `oc deploy --latest`
3. `oc scale dc/name --replicas=2`
4. `oc deploy --latest`
5. `oc scale dc/name --replicas=1`

Replicas on the RC and DC remain stuck at 2. I haven't done any deep debugging yet. Sorry for the confusion!